src/main/java/org/broadinstitute/hellbender/Main.java

```diff
@@ -282,7 +294,15 @@ private static CommandLineProgram extractCommandLineProgram(final String[] args,
                 if (simpleNameToClass.containsKey(args[0])) {
                     final Class<?> clazz = simpleNameToClass.get(args[0]);
                     try {
-                        return (CommandLineProgram) clazz.newInstance();
+                        if (clazz.getName().startsWith("picard.")) {
+                            // wrap the PicardCommandLinePrograms in a PicardCommandLineProgramExecutor
+                            @SuppressWarnings("unchecked")
+                            Class<? extends picard.cmdline.CommandLineProgram> picardClazz =
+                                    (Class<? extends picard.cmdline.CommandLineProgram>) clazz;
+                            return new PicardCommandLineProgramExecutor(picardClazz.newInstance());
```

Comment:
The common interface in Barclay will reduce the complexity here and keep the previous code untouchable...